src/nic.c

```diff
@@ -261,28 +261,23 @@ static int nic_poll(struct pg_brick *brick, uint16_t *pkts_cnt,
 static int nic_init_ports(struct pg_nic_state *state, struct pg_error **errp)
 {
 	int ret;
+	struct rte_eth_dev_info dev_info;
+	static struct rte_eth_txconf tx_conf;
 	struct rte_mempool *mp = pg_get_mempool();
-	static const struct rte_eth_conf port_conf = {
+
+	static struct rte_eth_conf port_conf = {
 		.rxmode = {
 			.split_hdr_size = 0,
-			/**< Header Split disabled */
-			.header_split   = 0,
-			/**< IP checksum offload disabled */
-			.hw_ip_checksum = 0,
-			/**< VLAN filtering disabled */
-			.hw_vlan_filter = 0,
-			/**< Jumbo Frame Support disabled */
-			.jumbo_frame    = 0,
-			/**< CRC stripped by hardware */
-			.hw_strip_crc   = 0,
+			.max_rx_pkt_len = ETHER_MAX_LEN,
 		},
 		.txmode = {
 			.mq_mode = ETH_MQ_TX_NONE,
 		},
 	};
-	static const struct rte_eth_txconf tx_conf = {
-		.txq_flags = 0,
-	};
+
+	rte_eth_dev_info_get(state->portid, &dev_info);
+	if (dev_info.tx_offload_capa & DEV_RX_OFFLOAD_CHECKSUM)
+		port_conf.txmode.offloads |= DEV_RX_OFFLOAD_CHECKSUM;
```

Comment:
If this work, it's because of luck:
you are using a rx flag for  `port_conf.txmode`,
You should use TX falgs
or better, I guess, you can just do
``` port_conf.txmode.offloads = dev_info.tx_offload_capa ```